### ISSUE TITLE
Error: Colon expected at position...

### ISSUE THREAD

**Issue by ta0s** [josdejong/jsonrepair#89]
Hi there.

When running jsonrepair from CLI (using [laravel homestead](https://github.com/laravel/homestead) 14.2.2) I get "Error: Colon expected at position" for any JSON entered which requires repair. Error does not fire when JSON is valid. 

I've tested multiple broken JSON in other tools using this library (https://josdejong.github.io/jsonrepair/) and it works every time.

Is there a known CLI issue? Is the JSON processed in some other way when not using CLI?

Cheers.

**Comment by josdejong** [comment]
Thanks for reporting. The CLI just works as far as I can see. 

How can I reproduce your issue?

**Comment by alita-moore** [comment]
I am having the same problem, here's a valid string:
```json
{
  "events": [
    {
      "type": "health journey",
      "context": [
        {
          "quote": "I've been home sick today but I don’t feel too terrible.",
          "interpretation": "The author is currently experiencing a mild illness.",
          "type": "situational"
        },
        {
          "quote": "I missed several meetings, some of which are gone forever and others that will need to be rescheduled. I also missed a lunch planned with a friend, and I hope to connect with her on Wednesday instead.",
          "interpretation": "The author missed important events due to their illness.",
          "type": "situational"
        },
        {
          "quote": "I read and napped today, and since I’d just finished my book, “The Girl Who Kicked the Hornet’s Nest,” things were kind of boring. I haven’t been to my Mom’s house lately so there weren’t any People magazines, either. That’s one reason to go to work. To avoid boredom. Teehee.",
          "interpretation": "The author spent their day at home reading and napping, but found it boring without new reading material. They also make a lighthearted comment about going to work to avoid boredom.",
          "type": "situational",
          "literaryDevice": "humorous"
        },
        {
          "quote": "The book club at the university is going to read “The Reliable Wife,” which I read several months ago. So I’ll need to at least skim it, if not read it again entirely, sometime soon.",
          "interpretation": "The author is a member of a book club and will need to revisit a book they have already read.",
          "type": "social"
        }
      ],
      "emotions": [
        {
          "type": "boredom",
          "intensity": "medium",
          "context": [
            {
              "quote": "I read and napped today, and since I’d just finished my book, “The Girl Who Kicked the Hornet’s Nest,” things were kind of boring. I haven’t been to my Mom’s house lately so there weren’t any People magazines, either. That’s one reason to go to work. To avoid boredom. Teehee.",
              "interpretation": "The author is feeling bored due to lack of new reading material and activities.",
              "type": "situational",
              "literaryDevice": "humorous"
            }
          ]
        }
      ],
      "keywords": [
        "illness",
        "missed meetings",
        "rescheduling",
        "missed lunch",
        "boredom",
        "book club"
      ]
    }
  ],
  "summary": "The author is currently experiencing a mild illness and missed important events as a result. They spent their day at home reading and napping, but found it boring without new reading material. They are a member of a book club and will need to revisit a book they have already read.",
  "keywords": [
    "illness",
    "missed meetings",
    "rescheduling",
    "missed lunch",
    "boredom",
    "book club"
  ]
}
```

but it errors out
<img width="1076" alt="Screenshot 2023-04-11 at 9 05 18 AM" src="https://user-images.githubusercontent.com/26529820/231188609-a73febfd-830d-468f-94bd-8a72b5d07639.png">
@josdejong 

Also, maybe that error message could be clearer?

**Comment by josdejong** [comment]
Thanks, now I'm able to reproduce this issue. It boils down to `jsonrepair` being a bit too greedy, here a minimal example:

```json
{
  "text": "Rounded “quotes” give an issue"
}
```
This is wrongly taking the first rounded quote before the word `quotes` as end of the string. I'll fix that by making the library a bit more conservative.

**Comment by alita-moore** [comment]
for those that may be facing this issue and the release isn't out yet, here's what I did (i.e. I just went with JSON.parse or jsonrepair, whichever worked first)

import {jsonrepair, JSONRepairError} from "jsonrepair";
import _ from "lodash";

function removeNilValues(obj) {
    return _.cloneDeepWith(obj, (value) => {
        if (_.isArray(value)) {
            const filtered = _.filter(value, (item) => !_.isNil(item));
            return filtered.map((item) => {
                if (_.isObject(item)) {
                    return removeNilValues(item);
                }
                return item;
            });
        }
    });
}

function replaceEmptyStrings(obj) {
    return _.cloneDeepWith(obj, (value) => {
        if (_.isArray(value)) {
            return _.map(value, (item) => {
                if (_.isString(item) && item.length === 0) {
                    return null;
                }
                if (_.isObject(item)) {
                    return replaceEmptyStrings(item);
                }
                return item;
            });
        }
    });
}

export const parseJSON = (json: string): any => {
    try {
         const parsed = removeNilValues(replaceEmptyStrings(JSON.parse(jsonrepair(json))));
         if (_.isNil(parsed.summary)) {
             parsed.summary = ""
         }
        return parsed;
    } catch (e) {
        if (e instanceof JSONRepairError) {
            try {
                // Try to parse the original JSON to workaround issues with jsonrepair
                const parsed = removeNilValues(replaceEmptyStrings(JSON.parse(json)));
                if (_.isNil(parsed.summary)) {
                    parsed.summary = ""
                }
                return parsed;
            } catch (_e: any) {
                console.log('failed to parse w/o jsonrepair',_e?.message);
            }
        }
        throw e
    }
}

**Comment by josdejong** [comment]
Thanks for sharing Alita. The fix has been published yesterday already in [v3.0.3](https://github.com/josdejong/jsonrepair/blob/develop/CHANGELOG.md#303-2023-04-17), can you give that a try?
